src/main/java/facade/UserFacade.java

```diff
@@ -7,165 +7,144 @@
-   * @Post Verifies that user exists, validates password, sets user to online, returns user cookie if it does exist.
-   * @return User cookie if user exists and password is verified.
-   * @throws UserNotFoundException if user is not found.
-   * @throws InvalidPasswordException if password is not verified.
-   */
-  public String login(String username, String password) throws UserNotFoundException, InvalidPasswordException, NoSuchAlgorithmException {
-
-      LOGGER.info("UserFacade.login");
-      User tempUser = userDao.getUser(username);
-      if(tempUser.getPassword() != password){
-          throw new InvalidPasswordException("Invalid Password");
-      }else{
-          tempUser.setOnline();
-          userDao.updateUser(tempUser);
-      }
-
-      return makeUserCookie(username);
-
-  }
-
-  /**
-   *
-   * @Pre User with matching username already exists.
-   * @Post Sets user to offline.
-   * @throws InvalidUserCookieException if the cookie is invalid.
-   */
-  public void logout(String username, String cookie) throws InvalidUserCookieException, UserNotFoundException{
-
-      LOGGER.info("UserFacade.logout");
-      if(!validateUserCookie(cookie)){
-          throw new InvalidUserCookieException();
-      }
-      User tempUser = userDao.getUser(username);
-      tempUser.setOffline();
-      userDao.updateUser(tempUser);
-  }
-
-  /**
-   * Finds matching user in DB and replaces it with the new User info.
-   *
-   * @Pre User is non null.
-   * @Post User with matching username will be updated but win record won't be affected.
-   * @throws UserNotFoundException if user does not exist.
-   * @throws InvalidUserCookieException if the cookie is invalid.
-   *
-   */
-  public void updateUser(User user, String cookie) throws UserNotFoundException, InvalidUserCookieException {
-
-      LOGGER.info("UserFacade.updateUser");
-      if(!validateUserCookie(user.getUsername())){
-          throw new InvalidUserCookieException();
-      }
-      userDao.updateUser(user);
-  }
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserFacade.class);
+    private UserDAO userDao;
+    private CookieManager cookieManager;
+
+    public UserFacade(){
+        userDao = UserDAO.getInstance();
+        cookieManager = new CookieManager();
+    }
+    public UserFacade(UserDAO userDAO){
+        this.userDao = userDAO;
+        cookieManager = new CookieManager();
+    }
+    public UserFacade(UserDAO userDAO, CookieManager cookieManager){
+        this.userDao = userDAO;
+        this.cookieManager = cookieManager;
+    }
+
+
+
+
+
+    /**
+     * @Pre None.
+     * @return List<String> of all usernames currently registered.
+     */
+    public List<String> getAllUsers() {
+        LOGGER.info("UserFacade.getAllUsers");
+        //TODO: get all users from DAO then just get the usernames and return them
+        List<String> usernames = new ArrayList<String>();
+        for(User u: userDao.getAllUsers()){
```

Comment:
You may want to have the `getAllUsers` call outside of the loop. This may call it over and over.